cloud/aws/aws_s3.cc

```diff
@@ -464,6 +465,13 @@ Status S3WritableFile::CopyManifestToS3(uint64_t size_hint, bool force) {
           "[s3] S3WritableFile made manifest %s durable to "
           "bucket %s bucketpath %s.",
           fname_.c_str(), s3_bucket_.c_str(), s3_object_.c_str());
+
+      // If cloud stats are present, record the manifest write and its latency in millis.
+      auto stats = env_->cloud_env_options.cloud_statistics;
+      if (stats) {
+          stats->recordTick(NUMBER_MANIFEST_WRITES, 1);
+          stats->measureTime(MANIFEST_WRITES_TIME, env_->s3client_result_.micros / 1000);
```

Comment:
Could you just measure time here instead of piggy-backing on the time measure in S3 callback? That would make the code much easier to reason about (no need for thread-local, wrapping the callback, etc). Also, you already have the start time of the operation recorded in line 455, so adding an extra time call shouldn't be a big concern.